### DISCUSSION TITLE
Data not returned in Rails app

### DISCUSSION THREAD

**Discussion by phedinkus** [mongodb/mongoid#5872] [Q&A]
I am running MongoDB in a container. I loaded the db with data from our staging env. I can see the data in the `mongosh` but when I connect via my Rails app the data is not visible (ie. `User.count` returns 0). I created a user through the Rails console and it was added to the database. In fact I can find that user in `mongosh`. I created a new `Mongo::Client` in my Rails console using the same options as the `Monogoid::Clients.default` and that client will return all the users in the data including the new one I created. What is different about how Mongoid instantiates the `Mongo::Client` that is preventing me from accessing all the data in the collection? If I call count on the collection from the client this way it returns the right number. 

![Screenshot 2024-09-26 at 14 11 31](https://github.com/user-attachments/assets/032496a8-ff7e-4e1b-920d-567de4686a29)

**Comment by phedinkus** [comment]
OMG, I realized the answer while reading through the `count` source code. Default Scopes!!!! We have an old field we barely use anymore that sets an associated record id from a config file. That record is different in staging than development. We were filtering them out. Holy moly did that wreck my brain.

**Reply by comandeo-mongo** [reply]
Default scopes strikes again! @phedinkus glad to hear that the issue was sorted out.
